### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by Have-Samuel**
In connecting to API: 
- Used the concept of state management.
- Used store, actions, and reducers in React.
- Connected React and Redux.
- Connected an existing API via Redux and use the data to display something in a UI.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,35 +1,63 @@
...
-      return { ...state, books: [...state.books, action.payload] };
-    }
-    case REMOVE_BOOK: {
-      const filteredBooks = state.books.filter(
-        (book) => book.id !== action.payload,
-      );
-      return { ...state, books: [...filteredBooks] };
-    }
+    case ADD_BOOK:
+      return [...state, ...action.payload];
```
Comment (mikethreels) [inline src/redux/books/books.js]: - [ ] you are using the spread operator as if you're adding to an array but the initialState is an object that's why the state doesn't get updated, what you can do to fix this is to change the initial state to an array and `return [...state, action.payload]`

File: src/redux/books/books-api.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export const fetchBooks = async () => {
+  try {
+    const res = await fetch(
+      BooksUrl,
+    );
+    const data = await res.json();
+    const books = [];
+    Object.entries(data).forEach((book) => book.push({ ...book[1][0], item_id: book[0] }));
+    return books;
```
Comment (mikethreels) [inline src/redux/books/books-api.js]: - [ ] you are returning the books array here but this always will be an empty array as set in line 10

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Have, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️ 
- [ ] the project is not working as expected the user is not able to add a book to the list, I've left a few inline comments but keep in mind that updating those suggestions might give you other errors, I advise you when testing to put some console.log() in your code to see what it is returning.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**